test/services/disableUserService.test.js

```diff
@@ -0,0 +1,183 @@
+import chai from 'chai'
+import chaiAsPromised from 'chai-as-promised'
+import { describe, before, it } from 'mocha'
+import Database from '../../src/services/gateway/database'
+import DisableUserService from '../../src/services/disableUserService'
+import { User } from './usersFactory'
+
+chai.use(chaiAsPromised)
+const expect = chai.expect
+
+describe('disableUserService', () => {
+  describe('blockUser', () => {
+    const maleForFriends = new User().male().likesFriends().get()
+    const femaleForFriends = new User().female().likesFriends().get()
+
+    before(() => {
+      const users = {
+        [maleForFriends.Uid]: maleForFriends,
+        [femaleForFriends.Uid]: femaleForFriends
+      }
+      Database('users').set(users)
+    })
+
+    describe('No disabled users', () => {
+      before(() => {
+        Database('disabledUsers').set({})
+      })
+
+      it('should save the user', () => {
+        return DisableUserService().blockUser(maleForFriends.Uid).then(() => {
+          return Database('disabledUsers').child(maleForFriends.Uid.toString()).once('value').then(result => {
+            expect(result.exists()).to.be.true
+          })
+        })
+      })
+    })
+
+    describe('There is a user disabled', () => {
+      before(() => {
+        const disabledUsers = {
+          [femaleForFriends.Uid]: true
+        }
+        Database('disabledUsers').set(disabledUsers)
+      })
+
+      it('should save the user', () => {
+        return DisableUserService().blockUser(maleForFriends.Uid).then(() => {
+          return Database('disabledUsers').child(maleForFriends.Uid.toString()).once('value').then(result => {
+            expect(result.exists()).to.be.true
+          })
+        })
+      })
+
+      it('femaleUser should still be disabled', () => {
+        return Database('disabledUsers').child(femaleForFriends.Uid.toString()).once('value').then(result => {
+          expect(result.exists()).to.be.true
+        })
+      })
+    })
+  })
+
+  describe('unblockUser', () => {
+    const maleForFriends = new User().male().likesFriends().get()
+    const femaleForFriends = new User().female().likesFriends().get()
+
+    describe('No disabled users', () => {
+      before(() => {
+        Database('disabledUsers').set({})
+      })
+
+      it('should reject promise', () => {
+        return DisableUserService().unblockUser(maleForFriends.Uid)
+          .then(() => {
+            return Promise.reject(new Error('Expected method to reject.'))
+          })
+          .catch(() => {
+            return true
+          })
+      })
+    })
+
+    describe('There is a user disabled', () => {
+      before(() => {
+        const disabledUsers = {
+          [femaleForFriends.Uid]: true
+        }
+        Database('disabledUsers').set(disabledUsers)
+      })
+
+      it('should deleted the user', () => {
```

Comment:
should delete the user from disabledUsers

Reply:
👍